### PR TITLE
Support for read-only archived project exception

### PR DESCRIPTION
**Pull request by Raalsky**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [x] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/common/exceptions.py [hunk 1]
```diff
@@ -408,3 +408,18 @@ def __init__(self, path):
...
+    def __init__(self):
+        message = """
+{h1}
+----WritingToArchivedProjectException-----------------------------------------------------------------------
+{end}
+You're trying to write to a project that was archived.
+
+Mark the project as active again or use "read-only" mode to read from it.
+
+{correct}Need help?{end}-> https://docs.neptune.ai/getting_help
```
Comment (normandy7) [inline src/neptune/common/exceptions.py:423]:
```suggestion
Set the project as active again or use mode="read-only" at initialization to fetch metadata from it.

{correct}Need help?{end}-> https://docs.neptune.ai/help/error_writing_to_archived_project/
```

**Review by normandy7 — CHANGES_REQUESTED** [review]
I'll publish a help page for this :)

File: src/neptune/common/backends/utils.py [hunk 2]
```diff
@@ -45,14 +52,42 @@
...
 _logger = logging.getLogger(__name__)
 
 MAX_RETRY_TIME = 30
 retries_timeout = int(os.getenv(NEPTUNE_RETRIES_TIMEOUT_ENV, "60"))
 
 
+def ensure_json_body(response: "RequestsResponseAdapter") -> Dict[str, Any]:
+    try:
+        return response.json() or dict()
+    except Exception:
```
Comment (AleksanderWWW) [inline src/neptune/common/backends/utils.py:70]: I think that we can make this exception more precise

https://github.com/psf/requests/blob/6e5b15d542a4e85945fd72066bb6cecbc3a82191/requests/models.py#L948

File: src/neptune/common/backends/utils.py [hunk 3]
```diff
@@ -45,14 +52,42 @@
...
+        return {}
+
+
+def handle_json_errors(
+    response,
```
Comment (AleksanderWWW) [inline src/neptune/common/backends/utils.py:75]: Can we have a type hint for this argument? Plus a type hint for what the function returns?

**Review by AleksanderWWW — COMMENTED** [review]
Is it worth it to also add a unit test for `ensure_json_body` and/or `handle_json_errors`?

File: src/neptune/common/backends/utils.py (hunk 2 above)
Comment (Raalsky) [reply src/neptune/common/backends/utils.py]: Sure. Thanks! Done

File: src/neptune/common/backends/utils.py (hunk 3 above)
Comment (Raalsky) [reply src/neptune/common/backends/utils.py]: Sure. Done

**Review by normandy7 — APPROVED** [review]

**Review by AleksanderWWW — APPROVED** [review]

**Comment by Raalsky** [comment]
> Is it worth it to also add a unit test for `ensure_json_body` and/or `handle_json_errors`?

Sure. Done I think

**Review by AleksanderWWW — APPROVED** [review]

**Pull request merged**